dashboard/lti_new.py

```diff
@@ -234,6 +235,14 @@ def extract_launch_variables_for_tool_use(request, message_launch):
         password = ''.join(random.sample(string.ascii_letters, settings.RANDOM_PASSWORD_DEFAULT_LENGTH))
         user_obj = User.objects.create_user(username=username, email=email, password=password, first_name=first_name,
                                             last_name=last_name)
+
+    # Add username into the MyLA User table, since the data was not pulled in from cron job
+    user_id = CANVAS_DATA_ID_INCREMENT + int(canvas_user_id)
+    try:
+        MyLAUser.objects.filter(user_id=user_id).update(sis_name = username)
+    except MyLAUser.DoesNotExist:
+        logger.warn(f'Cannot find MyLA user {username}')
```

Comment:
Do we want to log a warning here? I think it's expected that some users will launch the tool before we have pulled cron data for them. It may just create log noise.

You could do this a little differently and not use a try/except.

```
enrollment_qs =  MyLAUser.objects.filter(user_id=user_id)
if enrollment_qs.exists():
    enrollment_qs.update(sis_name=username)
```

Reply:
Replaced with the suggested block.